themes/developer/developer.theme.sh

```diff
@@ -0,0 +1,214 @@
+  _omb_util_put "$value"
+}
+
+function __bobby_clock {
+  _omb_util_put "$(clock_prompt) "
+
+  if [[ $THEME_SHOW_CLOCK_CHAR == true ]]; then
+    _omb_util_put "$(clock_char) "
+  fi
+}
+
+function _omb_theme_developer_get_node_version {
+  local val_node=$(node --version)
+  if _omb_util_command_exists nvm; then
+    _omb_theme_developer_return_delimited "node" "nvm $val_node"
+  else
+    # Si nvm no está instalado, utilizar "njs"
+    _omb_theme_developer_return_delimited "node" "njs $val_node"
+  fi
+}
+
+function _omb_theme_developer_get_go_version {
+  local val_go=$(go version | cut -d ' ' -f 3 | cut -d 'o' -f 2)
+  _omb_theme_developer_return_delimited "go" "go $val_go"
+}
+
+function _omb_theme_developer_get_ruby_version {
+  local val_rb=$(ruby --version | cut -d ' ' -f 2)
+  if _omb_util_command_exists rvm; then
+    _omb_theme_developer_return_delimited "ruby" "rvm $val_rb"
+  else
+    # Si nvm no está instalado, utilizar "njs"
+    _omb_theme_developer_return_delimited "ruby" "rb $val_rb"
+  fi
+}
+
+function _omb_theme_developer_get_py_version {
+  local val_py=$(python --version | cut -d ' ' -f 2)
+  if _omb_util_command_exists conda; then
+    local condav=$(conda env list | awk '$2 == "*" {print $1}')
+    _omb_theme_developer_return_delimited "python" "conda<$condav> $val_py"
+  else
+    # Si nvm no está instalado, utilizar "njs"
+    _omb_theme_developer_return_delimited "python" "py $val_py"
+  fi
+}
+
+function _omb_theme_developer_OPi5p_Temp {
+  local temp_opi5p=$(< /sys/class/thermal/thermal_zone4/temp)
+  local temp_in_c=$((temp_opi5p / 1000))
+  _omb_util_put "$temp_in_c"
+}
+
+function _omb_theme_developer_genericLinuxTemp {
+  local temp_lnx=$(< /sys/class/thermal/thermal_zone0/temp)
+  local temp_in_c=$((temp_lnx / 1000))
+  _omb_util_put "$temp_in_c"
+}
+
+# if is a specific platfor use spacific configuration otherwise use default
+# linux configuration.
+function _omb_theme_developer_currentPlatform {
+  # 2 ways to detect the platform 1) use a env var 2) some scrapping from the
+  # current system info (this is bash so just linux is considered) env var is
+  # $PROMPT_THEME_PLATFORM
+  # TODO: this is a first basic implementation this could be better but for now
+  # is ok
+  local platform_according_env=$PROMPT_THEME_PLATFORM
+
+  # if opi5 -> search for rk3588 tag in kernel and ...
+  local opi5p_kernel_tag=$(uname --kernel-release | cut -d '-' -f 3)
+
+  if [[ $platform_according_env == OPI5P || $opi5p_kernel_tag == rk3588 ]]; then
+    _omb_util_put OPI5P
+  else
+    _omb_util_put linux
+  fi
+}
+
+function _omb_theme_developer_cpu_load {
+  # Ejecutar el comando top en modo batch, filtrar por el nombre de usuario
+  # actual y almacenar la salida en la variable 'output'
+  local output=$(top -b -n 1 -u $USER | grep "Cpu(s)")
+
+  # Extraer el porcentaje de carga de la CPU excluyendo el estado "idle" usando
+  # awk
+  local cpu_load=$(echo "$output" | awk '{print 100.0-$8}' | cut -d '.' -f 1)
```

Comment:
This line seems to fail when the current CPU usage is `100.0%`, in which case the CPU usage is connected to the previous field, and `$8` points to the next field. This is an example output:

```
%Cpu(s):  0.0 us,  0.0 sy,  0.0 ni,100.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
```

where `$7` and `$8` were supposed to be `ni,` and `100.0`, respectively, but the actual results of `$7` and `$8` would become `ni,100.0` and `id`, respectively. We should split the line with `,` and pick the field corresponding to `id`.

Reply:
c3168eb